src/Sentry/SentryOptions.cs

```diff
@@ -1324,6 +1325,18 @@ public SentryOptions()
         );
 
         NetworkStatusListener = new PollingNetworkStatusListener(this);
+
+#if NET8_0_OR_GREATER
+        // If source generators are available, we can detect the user's build configuration and use this
+        // to infer whether to enable debug mode by default.
+        // See: https://develop.sentry.dev/sdk/expected-features/#auto-debug-mode
+        string? buildConfig = null;
+        BuildProperties.Values?.TryGetValue("Configuration", out buildConfig);
+        if (buildConfig == "Debug")
```

Comment:
question: Case-sensitivity?

I wonder if this string equality check should be case insensitive 🤔

```CSharp
buildConfig.Equals("Debug", StringComparison.OrdinalIgnoreCase)
```

Reply:
I think it should, but we should configure the underlying dictionary to be case insensitive... that way we don't have to worry about it in method calls like this right?